lib/assetmanager.js

```diff
@@ -215,7 +216,37 @@
         var layerIdsWithComponents = [],
             compComponents = [],
             doc = this._document,
-            comps = doc._comps;
+            comps = doc._comps,
+            docMeta = this._document._generatorSettings[META_PLUGIN_ID];
+        
+        if (docMeta && docMeta.json) {
+            docMeta = JSON.parse(docMeta.json);
+        }
+        
+        if (docMeta && docMeta.metaEnabled) {
+            console.log("default components enabled");
+            
+            this._componentManager._defaultComponents = [];
+            //read the default layer spec
+            if (docMeta.scaleSettings) {
+                docMeta.scaleSettings.forEach(function (spec) {
+                    
+                    //make the spec as-expected for component manager
+                    
+                    if (!spec.folder) {
+                        spec.folder = ["/"];
+                    } else if (typeof spec.folder === "string") {
+                        spec.folder = [spec.folder];
+                    }
+                    if (!spec.file) {
+                        spec.file = "";
+                    }
+                    
+                    this._componentManager._defaultComponents.push(spec);
+                    this._componentManager.addMetaComponent(spec);
+                }, this);
+            }
+        }
```

Comment:
Is there supposed to be an else here? It look like right now you can have meta-data definitions added first, then layers and comps get added in after that. 

is the intended behavior that the meta switch is additive and both meta and layer names can / should run at the same time?